torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -595,11 +632,17 @@ TensorView* TensorView::cache_after() {
     setComputeAt(consumer, this_ca_pos, this_ca_pos);
     consumer->setComputeAt(this_ca_view, this_ca_pos, rel_ca_pos);
   } else {
+    bool cache_replayed = false;
     // Check users of this TV for computeAt for cache_after on inputs
     for (auto expr : fusion()->unordered_uses(consumer)) {
       for (TensorView* output :
            ir_utils::filterByType<TensorView>(expr->outputs())) {
         if (output->hasComputeAt()) {
+          if (!cache_replayed) {
+            // Completely transform consumer according to output
+            TransformReplay::replayPasC(consumer, output, -1);
+            cache_replayed = true;
+          }
```

Comment:
Can you comment why you're doing this?

Reply:
When scheduling softmax and normalization ops, we need to merge all non-reduction and reduction axes together and to place the merged reduction axis in the -1 position.

If we have a point-wise operation before normalization, we don't apply those necessary transformations to the cache_after tensor. We use `TransformReplay::replayPasC` to apply all the transformations from the input's consumers.

Reply:
Sorry, I don't understand why that needs to be done as part of `cache_after`. I'm not 100% sure if we really need this part of `cache_after` (see #483). 

Is this merely for convenience? Or is it required for correctness?

Reply:
> Do you not want to support calling cache-after and cache-before after creating computeAt structure? If so, why not?
> 

I haven't thought about it. Why are you asking this?

Reply:
Then what are you referring to with issue [#483](https://github.com/csarofeen/pytorch/issues/483)? 

It is only an issue if apply cache-after after setting computeAt.

Reply:
I suspect that #483 was confusing because the line numbers need to be updated. It was originally referring to the `else` part of `if (hasComputeAt())`. https://github.com/csarofeen/pytorch/blob/20_10_20_devel/torch/csrc/jit/codegen/cuda/tensor_view.cpp#L598-L610